controllers/watcher_controller.go

```diff
@@ -21,25 +21,32 @@ import (
 	"errors"
 	"fmt"
 
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime"
+	"github.com/kyma-project/lifecycle-manager/pkg/log"
+	"github.com/kyma-project/lifecycle-manager/pkg/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/kyma-project/lifecycle-manager/api/v1beta1"
 	"github.com/kyma-project/lifecycle-manager/pkg/istio"
-	"github.com/kyma-project/lifecycle-manager/pkg/log"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
 const (
 	watcherFinalizer = "operator.kyma-project.io/watcher"
 )
 
-var ErrRestConfigIsNotSet = errors.New("reconciler rest config is not set")
+var (
+	ErrRestConfigIsNotSet = errors.New("reconciler rest config is not set")
+	errRemovingFinalizer  = errors.New("error removing finalizer")
```

Comment:
if those variables are only used in package, rename them all start with lower case.